### ISSUE TITLE
Module not found can’t resole storybook-addon-headle/register 

### ISSUE THREAD

**Issue by Vykuntarao-n** [ArrayKnight/storybook-addon-headless#20]
I am following you medium page and trying to register storybook-addon-headless I am getting the module not found error when I start the storybook.

ERROR in multi ./node_modules/@storybook/core/dist/server/common/polyfills.js ./.storybook/manager.js ./node_modules/@storybook/core/dist/client/manager/index.js ./node_modules/@storybook/addon-essentials/node_modules/@storybook/addon-docs/dist/register.js ./node_modules/@storybook/addon-actions/dist/register.js ./node_modules/@storybook/addon-essentials/node_modules/@storybook/addon-backgrounds/dist/register.js ./node_modules/@storybook/addon-viewport/dist/register.js ./node_modules/@storybook/addon-knobs/dist/register.js ./node_modules/@storybook/addon-a11y/dist/register.js ./node_modules/@storybook/addon-controls/dist/register.js ./node_modules/storybook-addon-jsx/register.js ./node_modules/@storybook/addon-toolbars/dist/register.js storybook-addon-headless/register ./node_modules/@storybook/addon-storysource/register.js ./.storybook/generated-refs.js

Module not found: Error: Can't resolve 'storybook-addon-headless/register' in 'C:\workspace\storybook'

@ multi ./node_modules/@storybook/core/dist/server/common/polyfills.js ./.storybook/manager.js ./node_modules/@storybook/core/dist/client/manager/index.js ./node_modules/@storybook/addon-essentials/node_modules/@storybook/addon-docs/dist/register.js ./node_modules/@storybook/addon-actions/dist/register.js ./node_modules/@storybook/addon-essentials/node_modules/@storybook/addon-backgrounds/dist/register.js ./node_modules/@storybook/addon-viewport/dist/register.js ./node_modules/@storybook/addon-knobs/dist/register.js ./node_modules/@storybook/addon-a11y/dist/register.js ./node_modules/@storybook/addon-controls/dist/register.js ./node_modules/storybook-addon-jsx/register.js ./node_modules/@storybook/addon-toolbars/dist/register.js storybook-addon-headless/register ./node_modules/@storybook/addon-storysource/register.js ./.storybook/generated-refs.js main[12]

module.exports = {
    addons: ['storybook-addon-headless/register'],
}

**To Reproduce**

1.   Follow the steps from

https://medium.com/@arrayknight/how-to-get-real-data-into-storybook-8915f5371b6

**Expected behavior**
Application should start without any error 

-  storybook 6
- Windows OS
- storybook-addon-headless 2.0.1
- 
package.json

{

  "name": " storybook",

  "version": "3.1.52",

  "main": "src/components/index.ts",

  "publishConfig": {

    "registry": "

  },

  "scripts": {

    "lint:eslint": "eslint \"./**/*.{js,jsx,ts,tsx}\" --fix",

    "lint:prettier": "prettier \"./**/*.{json,yml,css,scss,md}\" --write",

    "lint": "npm-run-all lint:*",

    "build": "yarn build-storybook -o ./public",

    "storybook": "start-storybook -p 9009 -s public",

    "serve": "npx http-server ./public",

    "suite": "react-scripts test --coverage",

    "start": "npm-run-all storybook",

    "test": "npm-run-all lint suite build",

    "add-component": "yarn generate component",

    "add-documentation": "yarn generate documentation",

    "generate": "yarn plop --plopfile ./.storybook/plopfile.js"

  },

  "jest": {

    "collectCoverageFrom": [

      "**/*.{ts,tsx}",

      "!dist/**",

      "!**/*.stories.ts*",

      "!**/index.ts"

    ],

    "coverageThreshold": {

      "global": {

        "statements": 68,

        "branches": 60,

        "functions": 68,

        "lines": 68

      }

    }

  },

  "husky": {

    "hooks": {

      "pre-commit": "lint-staged"

    }

  },

  "lint-staged": {

    "*.{js,jsx,ts,tsx}": [

      "eslint --fix",

      "git add"

    ],

    "*.{json,yml,css,scss,md}": [

      "prettier --write",

      "git add"

    ]

  },

  "browserslist": [

    ">0.2%",

    "not dead",

    "not ie <= 11",

    "not op_mini all"

  ],

  "repository": {

    "type": "git",

    "url": “”

  },

  "dependencies": {

    "@apollo/client": "^3.3.6",

    "@brightcove/react-player-loader": "^1.3.0",

    "@contentful/forma-36-tokens": "^0.9.2",

    "@contentful/rich-text-react-renderer": "^13.2.0",

    "@contentful/rich-text-types": "^13.1.0",

    "@emotion/hash": "^0.7.2",

    "@emotion/styled": "^10.0.27",

    "@fortawesome/fontawesome-svg-core": "^1.2.25",

    "@fortawesome/free-brands-svg-icons": "^5.11.2",

    "@fortawesome/free-regular-svg-icons": "^5.11.2",

    "@fortawesome/free-solid-svg-icons": "^5.11.2",

    "@fortawesome/react-fontawesome": "^0.1.4",

    "@storybook/addon-a11y": "6.1.7",

    "@storybook/addon-essentials": "6.1.7",

    "@storybook/addon-knobs": "^6.1.7",

    "@storybook/addon-storysource": "6.1.7",

    "@storybook/addons": "^6.1.10",

    "@storybook/react": "^6.1.7",

    "@types/enzyme": "^3.10.5",

    "@types/enzyme-adapter-react-16": "^1.0.6",

    "@types/file-saver": "~2.0.1",

    "@types/lodash": "^4.14.136",

    "@types/react-datepicker": "^2.8.1",

    "@types/react-select": "^3.0.19",

    "@types/react-tabs": "^2.3.1",

    "@types/react-test-renderer": "^16.9.0",

    "apollo-boost": "^0.4.9",

    "classnames": "^2.2.6",

    "copy-to-clipboard": "^3.2.1",

    "emotion": "^10.0.14",

    "enzyme": "^3.11.0",

    "enzyme-adapter-react-16": "^1.15.2",

    "eslint-plugin-prettier": "^3.1.0",

    "eslint-plugin-react-hooks": "^1.7.0",

    "eslint-plugin-sonarjs": "^0.5.0",

    "exceljs": "~4.1.1",

    "file-saver": "~2.0.2",

    "gatsby-background-image": "^0.8.15",

    "gatsby-image": "^2.2.37",

    "graphql": "^15.4.0",

    "i": "^0.3.6",

    "js-cookie": "^2.2.1",

    "linkifyjs": "^2.1.9",

    "lodash": "^4.17.15",

    "luxon": "^1.22.2",

    "moment-timezone": "^0.5.31",

    "moment-timezone-data-webpack-plugin": "^1.3.0",

    "node-sass": "^4.12.0",

    "npm": "^6.14.9",

    "plop": "^2.7.4",

    "prop-types": "^15.7.2",

    "react": "^17.0.1",

    "react-apollo": "^3.1.5",

    "react-datepicker": "^2.9.6",

    "react-dom": "^16.9.0",

    "react-html-parser": "^2.0.2",

    "react-lazyload": "^2.6.8",

    "react-page-visibility": "^5.1.0",

    "react-perfect-scrollbar": "^1.5.3",

    "react-player": "^1.8.0",

    "react-scripts": "^3.1.2",

    "react-scroll": "^1.7.11",

    "react-select": "^3.1.0",

    "react-slick": "^0.24.0",

    "react-table": "^6.10.0",

    "react-tabs": "^3.0.0",

    "react-test-renderer": "^16.9.0",

    "react-ticker": "^1.2.1",

    "react-toastify": "^5.4.0",

    "slick-carousel": "^1.8.1",

    "uniqid": "^5.0.3",

    "video.js": "^7.6.0"

  },

  "devDependencies": {

    "@babel/core": "^7.6.2",

    "@babel/plugin-transform-react-jsx-development": "^7.9.0",

    "@storybook/addon-backgrounds": "^6.1.7",

    "@storybook/addon-docs": "^6.1.7",

    "@storybook/theming": "^6.1.10",

    "@types/classnames": "^2.2.9",

    "@types/jest": "^24.0.13",

    "@types/js-cookie": "^2.2.6",

    "@types/linkifyjs": "^2.1.3",

    "@types/luxon": "^1.22.0",

    "@types/node": "^12.6.8",

    "@types/prop-types": "^15.7.3",

    "@types/react": "^16.8.23",

    "@types/react-dom": "^16.8.5",

    "@types/react-html-parser": "^2.0.1",

    "@types/react-intl": "^2.3.17",

    "@types/react-lazyload": "2.3.2",

    "@types/react-scroll": "^1.5.4",

    "@types/react-slick": "^0.23.3",

    "@types/react-table": "^6.8.5",

    "@types/storybook__react": "^5.2.1",

    "@types/uniqid": "^4.1.3",

    "@types/video.js": "^7.2.11",

    "@typescript-eslint/eslint-plugin": "^2.3.1",

    "@typescript-eslint/parser": "^2.3.1",

    "@utilitywarehouse/storybook-addon-prop-types": "^1.0.1",

    "babel-loader": "^8.2.2",

    "babel-plugin-react-docgen": "^4.2.1",

    "eslint-config-airbnb": "^17.1.1",

    "eslint-config-prettier": "^6.0.0",

    "eslint-import-resolver-typescript": "^1.1.1",

    "eslint-plugin-import": "^2.18.0",

    "eslint-plugin-jsx-a11y": "^6.2.3",

    "eslint-plugin-react": "^7.14.3",

    "husky": "^3.0.1",

    "lint-staged": "^8.1.7",

    "npm-run-all": "^4.1.5",

    "prettier": "^1.17.1",

    "react-ticker": "^1.2.1",

    "storybook-addon-headless": "^2.0.1",

    "storybook-addon-jsx": "^7.3.4",

    "typescript": "^3.6.3"

  },

  "resolutions": {

    "caniuse-lite": "1.0.30000977"

  }

}

**Comment by Mrostgaard** [comment]
Hey

I had the same issue. I got it working by registering the addon in the main.js file as such:

`
module.exports = {
  stories: ["../src/stories/**/*.stories.(ts|tsx|js|jsx|mdx)"],
  addons: [
    ....
    'storybook-addon-headless/dist/register',
    ....
  ],
};

`

**Comment by Vykuntarao-n** [comment]
🙏. was able to resolve the original error now I am getting the bellow error 

Cannot convert undefined or null to object

TypeError: Cannot convert undefined or null to object

    at keys (<anonymous>)

    at Function.keys (http://localhost:9009/vendors~main.db47f17a52e6d6f7f7e8.bundle.js:119044:24)

    at http://localhost:9009/vendors~main.db47f17a52e6d6f7f7e8.bundle.js:319747:22

    at renderWithHooks (http://localhost:9009/vendors~main.db47f17a52e6d6f7f7e8.bundle.js:273948:18)

    at updateFunctionComponent (http://localhost:9009/vendors~main.db47f17a52e6d6f7f7e8.bundle.js:275912:20)

    at updateSimpleMemoComponent (http://localhost:9009/vendors~main.db47f17a52e6d6f7f7e8.bundle.js:275850:10)

    at updateMemoComponent (http://localhost:9009/vendors~main.db47f17a52e6d6f7f7e8.bundle.js:275753:14)

    at beginWork$1 (http://localhost:9009/vendors~main.db47f17a52e6d6f7f7e8.bundle.js:277798:16)

    at HTMLUnknownElement.callCallback (http://localhost:9009/vendors~main.db47f17a52e6d6f7f7e8.bundle.js:257990:14)

    at Object.invokeGuardedCallbackDev (http://localhost:9009/vendors~main.db47f17a52e6d6f7f7e8.bundle.js:258039:16)

**Comment by Mrostgaard** [comment]
Yes that's the same error I get now. 'can't convert undefined to object'. I don't know why this is happening. I will let you know when I find a solution. Hopefully you will let me know too :-)

**Comment by ArrayKnight** [comment]
I've just published version 2.1.0 which should no longer have an issue with referencing `register` in the root. Also, my hope is that you would be able to register it is an addon without needing to specify `register` at all:

```
module.exports = {
    addons: ['storybook-addon-headless'],
    ...,
}
```

Please give that a try

**Comment by ArrayKnight** [comment]
I've confirmed that `2.1.0` fixes the register issue. And `2.1.1` which is being published now will fix the undefined object issue
